### ISSUE TITLE
Error executing a lua callback on TSUpdate

### ISSUE THREAD

**Issue by ojroques** [nvim-treesitter/nvim-treesitter#744]
- `nvim --version`: `NVIM v0.5.0-dev+917-g85bce4d13`
- nvim-lsp version(commit hash): `76f1162`
- `:checkhealth` result:
```
health#nvim_treesitter#check
========================================================================
## Installation
  - OK: `git` executable found.
  - OK: `cc` executable found.

## c parser healthcheck
  - OK: c parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `folds.scm` found.
  - WARNING: No `indents.scm` query found for c
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

health#nvim#check
========================================================================
## Configuration
  - OK: no issues found

## Performance
  - INFO: Build type: Debug
  - WARNING: Non-optimized (DEBUG) build. Nvim will be slower.
    - ADVICE:
      - Install a different Nvim package, or rebuild with `CMAKE_BUILD_TYPE=RelWithDebInfo`.
      - https://github.com/neovim/neovim/wiki/FAQ

## Remote Plugins
  - OK: Up to date

## terminal
  - INFO: key_backspace (kbs) terminfo entry: key_backspace=\177
  - INFO: key_dc (kdch1) terminfo entry: key_dc=\E[3~
```

- Operating system/version: Ubuntu 20.04 in WSL1
- Terminal name/version: Windows Terminal v1.5.3242.0
- `$TERM`: `xterm-256color`

### How to reproduce the problem from neovim startup
With this minimal config:
```lua
vim.cmd 'packadd paq-nvim'  # my plugin manager
require('paq-nvim').paq 'nvim-treesitter/nvim-treesitter'
```
Running `:TSUpdate` gives me this error:
```
[nvim-treesitter] [0/1] Compiling...
Error executing vim.schedule lua callback: ...qs/start/nvim-treesitter/lua/nvim-treesitter/install.lua:83: 
bad argument #1 to 'spawn' (string expected, got userdata)
```

I did make sure to use the latest neovim and nvim-treesitter so I'm not sure where it comes from. Any idea why that happens?

**Comment by vigoux** [comment]
I can't seem to reproduce on my end, are you sure you use the latest `nvim-treesitter` version ?

**Comment by theHamsta** [comment]
This suggests that `cc` has some non-lua value. Can you check its value at this line (with a `print(vim.inspect(cc))`)? Can you fix it by setting this value to `'gcc'`

https://github.com/theHamsta/nvim-treesitter/blob/528506ae2a713c31e2e3f275ac65568e9d2a169e/lua/nvim-treesitter/install.lua#L162

Does this also happen with `TSInstall` / `TSInstallSync`?

**Comment by ojroques** [comment]
I've checked the code and this is due to a very weird behavior I have on my machine:
```sh
:lua print(vim.fn.executable(vim.NIL))
1
```
So this means this function:
https://github.com/nvim-treesitter/nvim-treesitter/blob/3727be629ce1cac799ca740b95fe3aa484148018/lua/nvim-treesitter/shell_command_selectors.lua#L52-L54

returns the first value in `M.compilers`:
https://github.com/nvim-treesitter/nvim-treesitter/blob/3727be629ce1cac799ca740b95fe3aa484148018/lua/nvim-treesitter/install.lua#L14

which is `vim.NIL` since I did not set `CC`... And of course it makes the plugin crashes.

I managed to work around the issue by setting `CC=gcc`. I would very much like to know why `vim.NIL` is considered an executable though, any idea why?

**Comment by theHamsta** [comment]
`:lua print(vim.fn.executable(vim.NIL))` is 0 on my machine

**Comment by ojroques** [comment]
I've kept investigating and I found the issue. It turns out that the C function mapped to `executable()` which is [is_executable_in_path(const char *name, char **abspath)](https://github.com/neovim/neovim/blob/480515e4426865c37a42a452432b3463f0ab40a7/src/nvim/os/fs.c#L348) receives the string `"null"` in the `name` argument when we call `executable()` with `vim.NIL` or `v:null`. And it turns out that somehow I do have an executable called *null* on Windows in `C:\Windows\System32`:
![image](https://user-images.githubusercontent.com/23409060/101493254-1f545500-3966-11eb-89c9-e8a13403f95f.png).

After deleting it my issue was solved. But to me this is an unexpected behavior, the C function should receive an empty buffer instead of a valid string which could always resolve to an actual executable, even if called *null*. I will open an issue in Neovim.
